aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -762,6 +798,31 @@ private int createMultipartUploadRecords(String bucket, String key, File file, O
         return dbUtil.bulkInsertTransferRecords(valuesArray);
     }
 
+    private File writeInputStreamToFile(InputStream inputStream) throws IOException {
+        if (inputStream == null) {
+            throw new IllegalArgumentException("Invalid inputStream: " + inputStream);
+        }
+
+        // Saves the data as a file in the temporary directory
+        File file = File.createTempFile(TransferStatusUpdater.TEMP_FILE_PREFIX, ".tmp");
+        OutputStream outStream = new FileOutputStream(file);
+        try {
+            byte[] buffer = new byte[1024];
+            int bytesRead;
+            // Keep reading until reaches the end of the stream
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                outStream.write(buffer, 0, bytesRead);
+                outStream.flush();
+            }
+        } catch (IOException ioException) {
+            file.delete();
+            throw new IOException("Error writing the inputStream into a file.", ioException);
+        } finally {
+            outStream.close();
+        }
```

Comment:
I think you can simplify this by just moving `file.delete()` to the `finally` block and getting rid of the `catch` block.  This ensures that the File will be deleted, and an `IOException` will be thrown up to the the client.

```suggestion
        } finally {
            file.delete();
            outStream.close();
        }
```

Reply:
I think we cannot excecute `file.delete()` in `finally` block because we only want to delete a file when there is an exception occurs when writing into it. But if there is no exception, we want to keep that file. And delete it later when the transfer is completed.

Reply:
Ahh!  Of course, that's right.  Ignore this comment :)